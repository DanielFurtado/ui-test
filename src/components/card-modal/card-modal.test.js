
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react'
import '@testing-library/jest-dom';
import CardModal from '../card-modal/card-modal.component';

const data = {
  'id': 'abc123',
  'description': 'some description.',
  'alt_description': 'alternative description',
  'likes': 10,
  'user': {
    'name': 'Fred Mercury',
    'profile_image': 'https://www.some-link/images/profile-image-1'
  },
  'url': 'https://www.some-link/images/some-image-1'
};

describe(('Card Item'), () => {
  it('renders', () => {
    const component = renderer.create(
      <CardModal show={true} image={data} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays the user name', () => {
    const {getByText} = render(
      <CardModal show={true} image={data} />
    );
    expect(getByText('Preview')).toBeInTheDocument();
  });

  it('should close the modal', () => {
    const closeModalSpy = jest.fn();
    const { getByTestId } = render(
      <CardModal show={true} image={data} onClose={closeModalSpy} />
    );
    fireEvent.click(getByTestId('card-modal-backdrop'));
    expect(closeModalSpy).toHaveBeenCalled();
  });

  it('should stop propagation if clicked', () => {
    const stopPropagationSpy = jest.fn();
    const { getByTestId } = render(
      <CardModal show={true} image={data} onClick={stopPropagationSpy} />
    );
    fireEvent.click(getByTestId('card-modal'));
    expect(stopPropagationSpy).toHaveBeenCalledTimes(0);
  });
});
