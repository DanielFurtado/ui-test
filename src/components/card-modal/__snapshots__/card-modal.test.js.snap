// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Item renders 1`] = `
<div
  className="card-modal-backdrop"
  data-testid="card-modal-backdrop"
>
  <div
    className="card-modal"
    data-testid="card-modal"
    onClick={[Function]}
  >
    <div
      className="card-modal-header"
    >
      <h4
        className="card-modal-title"
      >
        Preview
      </h4>
      <button
        className="card-modal-close-btn"
        type="button"
      >
        <span
          className="icon-cross"
        />
        <span
          className="visually-hidden"
        >
          Close
        </span>
      </button>
    </div>
    <div
      className="card-modal-body"
    >
      <div
        className="card-modal-image-wrapper"
      >
        <img
          alt="alternative description"
          className="card-modal-image"
          loading="lazy"
          src="https://www.some-link/images/some-image-1.jpg"
        />
        <div
          className="card-modal-image-likes"
        >
          <span>
            <svg
              className="like-icon"
            >
              like.svg
            </svg>
          </span>
          10
        </div>
      </div>
      <div
        className="card-modal-info"
      >
        <div
          className="card-modal-profile-wrapper"
        >
          <div
            className="card-modal-profile-image"
          >
            <img
              className="card-profile-image"
              loading="lazy"
              src="https://www.some-link/images/profile-image-1.webp"
            />
            <div
              className="card-profile-username"
            >
              Fred Mercury
            </div>
          </div>
        </div>
        <div
          className="card-description"
        >
          some description.
        </div>
        <button
          aria-label="Download"
          className="button-primary"
          onClick={[Function]}
        >
          Download
        </button>
      </div>
    </div>
  </div>
</div>
`;
